src/Router.js

```diff
@@ -160,6 +161,10 @@ export default class Router {
 
       this.__currentCanonicalPath = canonicalPath
 
+      if (this.onRouteStart && mode !== 'replace') {
+        this.onRouteStart();
```

Comment:
I think it'd be good to supply some data to this function.  Like the `fromPath` and make this threads through properly with redirects, as well as the `startTime`.

Thoughts?